### PR TITLE
fix: dynamic text scaling

### LINKED ISSUE #850: Dynamic Text Sizing
# Background
We have had community contributors make a few passes (see #709 ) at dynamic text sizing and while the issue has been resolved, we think there may be a better css based solution (potentially using clamp) to solve this.

This issue can be picked up by anyone and we will happily tip anyone that has a clean solution to the problem and opens a PR that implements that and removes the old code paths from #836 

# Acceptance Criteria
- CSS based solution to dynamic text sizing in fields that expand as the number of digits grown
- Tested and working in all localizations
- bonus points and tip for a re-usable component that we can implement anywhere this type of text would be needed.

# Screenshots 
Current Solution Example:
image
Old issue that prompted the fix (see the clipping):
image

### PR DESCRIPTION
**Pull request by bronco-eth**
## Description

* fixes #850 – current `FiatInput` component still had same clipping issues on smaller screens
* Introducing basic dynamically resizable, reusable component `FlexibleInputContainer`

https://www.loom.com/share/934d4d4838d74148853d2baae7feb763

Note: Solution is still JS based, just reduced in complexity. CSS only solution would be quite challenging to pull off. For accurate scaling font size should depend on the text length which is accessible via JS only.

Good next step would be to make token input fields scalable as well.

## Notice

Before submitting a pull request, please make sure you have answered the following:

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?
- [x] Do all new and existing tests pass? Does the linter pass?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps, `cp .sample.env .env`, `yarn dev`
2. Type in a long number into the fiat currency input field.
3. Check how it works on different screen sizes.

### CODE REVIEW

**Review by cjthompson — APPROVED** [review]

**Comment by 0xean** [comment]
@bronco-eth - Nice work!  I will get a tip over to you for the work right now.

Also - `a dood next step would be to make token input fields scalable as well.` - totally agree! Do you have any interest in tackling that also?

If so, could you create a ticket and assign yourself to it? Happy to keep the FOX flowing your way for the help!

**Review by 0xean — APPROVED** [review]

**Comment by bronco-eth** [comment]
hey @0xean - thanks a lot! yes, i'll look into proposing a spec for the token fields.

File: src/components/FlexibleInputContainer/FlexibleInputContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      resizeObserver.observe(scaledContainerRef)
+
+      return () => resizeObserver.unobserve(scaledContainerRef)
+    }
+  }, [scaledContainer])
+
+  useEffect(updateScale, [props.value])
+
+  return (
+    <div style={{ fontSize: props.fontSize as string, ...wrapperStyles }}>
```
Comment (reallybeard) [inline src/components/FlexibleInputContainer/FlexibleInputContainer.tsx:65]: Can we use `<Box>` for these and not straight `<Div>`?

**Review by reallybeard — COMMENTED** [review]
Other than my question about using `<Box>` instead of `<Div>` LGTM

**Review by reallybeard — APPROVED** [review]

**Pull request merged**